File: C++/password/password.cpp
#include <iostream>
#include <queue>
using namespace std;

int main() {
   double rem = 1;
   double total = 0;

   int n;
   cin >> n;

   string trash;
   double curr;
   
   priority_queue<double> q;

   for (int i = 0; i < n; i++) {
      cin >> trash >> curr;
      q.push(curr);
   }

   for (int i = 0; i < n; i++) {
      curr = q.top();
      q.pop();
      total += rem;
      rem -= curr;
   }
   
   cout << total << endl;

   return 0;
}

File: C++/password/password.py
curr = list()
rem = 1.0
total = 0.0

n = int(input())

lis = list()

for x in range(n):
   curr = input().split()
   lis.append(float(curr[1]))

lis.sort()
#lis.reverse()
it = 0.0

for x in range(n):
   it = lis.pop()
   total += it*(x+1)
   rem -= it
   
print(total)